### DISCUSSION TITLE
[Exercise-1]: What constituties a PL/SQL block? Explain in your own words and give any examples.

### DISCUSSION THREAD

**Discussion by acrlakshman** [exaslate-courses/oracle-sql-plsql-b3#2] [General]
Reference: https://www.youtube.com/watch?v=S0XnhFOmCJ0&list=PLF6_5RI6ENIlDi3knzcQGUiFc5-WdA_1W&index=1&t=880s

**Comment by srideviadapa0511** [comment]
Blocks are considered as a unit of code .The blocks can be loops,stings,constants and conditional satements. A block is divided in to 3 parts such as "Declare" (where we declare any variables),"Begin"(where execution is done),exception (where we print any exception before executing them),"End"(where program ends).
DECLARE
age number (3) := 25;
BEGIN
DBMS_OUTPUT.PUT_LINE(AGE);
EXCEPTION
WHEN others THEN
DBMS_OUTOUT.PUT_LINE(sql code| | ' : ' | | sqlerrm);
END;
/

**Reply by acrlakshman** [reply]
Good work @srideviadapa0511 .

**Comment by Tharun1596** [comment]
we have four ~constituties~ *components* in pl/sql block that are
DECLARE: declare the variables in code.
BEGIN: actual code that we write to excute,
EXCEPTION: when ever we write wrong code we can do the exception.
END: Program e~m~*n*ds.

Here the sample example 
DECLARE
my friend name (NAME)  := sri;
BEGIN
DBMS_OUTPUT.PUT_LINE(NAME);
EXCEPTION
WHEN others THEN
DBMS_OUT~O~*P*UT.PUT_LINE(sql code| | ' : ' | | sqlerrm);
END;
/

OUTPUT: 
SRI

**Reply by acrlakshman** [reply]
@Tharun1596 Nice work. Please revise my edits. My corrections include, strikethrough and edits in italics.

**Reply by Tharun1596** [reply]
Okay sir thank you

**Comment by YaminiYadavv** [comment]
A PL/SQL (Procedural Language/ Structured Query Language) block is a collection of group statements, which were combined into a single unit. It has four components which Include: DECLARE, BEGIN, EXCEPTION, END;
**Declare**: In this section we can Declare the variables, constants and more inside the block.
**Begin**: This is an Execution part where we can also declare the variables, it is not mandatory to Declare the variables only in the declaration section we can declare them in the Begin part too. We can also have the assignment operators like (IF, ELSEIF, Loops, While Loop, For Loop conditional statements).
**Exception**: In any coding structure Exception is considered as Handling the Errors that may occur during the code.
**End**: It is considered as End of the code which should always have a symbol ';'.

Here is the example code:
DECLARE
phone number(10):= 1234567894;
BEGIN
DBMS_OUTPUT.PUT_LINE(PHONE);
EXCEPTION
WHEN others THEN
DBMS_OUTPUT.PUT_LINE(sqlcode || ' : ' || sqlerrm);
END;
/

**Comment by cm2024jsp** [comment]
A block is broken into three sections: "Declare" (where we declare any variables), "Begin" (where execution begins), "Exception" (where we report any exceptions before executing them), and "End" (where the programme stops).
